compositor_pipeline/src/pipeline.rs

```diff
@@ -39,9 +46,56 @@ pub trait PipelineInput: Send + Sync + 'static {
     fn new(queue: Arc<Queue>, opts: Self::Opts) -> Self;
 }
 
-pub struct Pipeline<Input: PipelineInput, Output: PipelineOutput> {
+pub struct OutputOptions<Receiver: PipelineOutputReceiver> {
+    pub receiver_options: Receiver::Opts,
+    pub encoder_settings: EncoderSettings,
+    pub resolution: Resolution,
+}
+
+pub struct PipelineOutput<Receiver: PipelineOutputReceiver> {
```

Comment:
This could be part of the Encoder implementation, if you want to keep them as a separate types I would:
- call this Encoder
- rename Encoder to LibavH264Encoder
- move this code to separate file (add as little as possible to the pipeline.rs)

Reply:
It seems very unintuitive to me that an `Encoder` would also take care of sending the encoded data out. I agree the naming is quite unfortunate here, but I don't like this suggestion either.